### PR TITLE
Add Busbar Chart

### PR DESCRIPTION
**Pull request by waldman**
Add busbar (https://github.com/busbar-io/busbar-server) Chart to allow installation through Helm.

### CODE REVIEW

File: incubator/busbar/templates/deployment.yaml
Comment (mattfarina) [inline incubator/busbar/templates/deployment.yaml]: The image names should not be hard coded. Here there is a separation between registry and image. Instead it should be the image (which can be a full URL) and the tag. For example (from coachroachdb):

https://github.com/kubernetes/charts/blob/5bff4b77c5c2cd40e244bfdba72b068248384745/stable/cockroachdb/values.yaml#L7-L8

File: incubator/busbar/.helmignore
Comment (mattfarina) [inline incubator/busbar/.helmignore]: The standard helmignore file has a bit more to it. For reference see https://github.com/kubernetes/charts/blob/5bff4b77c5c2cd40e244bfdba72b068248384745/stable/artifactory/.helmignore.

File: incubator/busbar/templates/deployment.yaml
Comment (mattfarina) [inline incubator/busbar/templates/deployment.yaml]: Deployments should be using `apps/v1beta1`

File: incubator/busbar/Chart.yaml
Comment (mattfarina) [inline incubator/busbar/Chart.yaml]: For a maintainers name we're looking for a GitHub ID. This way we have someone to notify when a PR or change comes it.

**Review by mattfarina — CHANGES_REQUESTED** [review]
Thank you for submitting this chart. I wasn't able to give this a full review but here is some feedback from a partial review.

File: incubator/busbar/templates/deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/deployment.yaml]: The `app` label should be `{{ template "busbar.name" . }}` on all resources the chart creates.

File: incubator/busbar/templates/deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/deployment.yaml]: After changing the `app` label as suggested above, you'll have to add the `release` label to labels selectors. Apply everywhere.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/busbar/values.yaml
Comment (mattfarina) [inline incubator/busbar/values.yaml]: The image and tag has a standard format we would like to see used.
```yaml
image:
  name: myapp
  tag: 1.2.3
  pullPolicy: IfNotPresent
```
You can read more about it in the review guide... https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#configuration

Comment (waldman) [reply incubator/busbar/values.yaml]: Thanks for the advise mate! =)
Changing it now! =P

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: image.name -> image.repository

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: IfNotPresent

File: incubator/busbar/templates/daemonset.yaml
Comment (unguiculus) [inline incubator/busbar/templates/daemonset.yaml]: The `app` label should be `{{ template "busbar.name" . }}`. Please fix this throughout the chart.

File: incubator/busbar/templates/deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/deployment.yaml]: Add `component` label.

File: incubator/busbar/templates/deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/deployment.yaml]: Add `component` label.

File: incubator/busbar/templates/deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/deployment.yaml]: One file per resource please.

File: incubator/busbar/templates/_helpers.tpl
Comment (unguiculus) [inline incubator/busbar/templates/_helpers.tpl]: Call it `busbar.mongodb.fullname`

File: incubator/busbar/templates/_helpers.tpl
Comment (unguiculus) [inline incubator/busbar/templates/_helpers.tpl]: Call it `busbar.redis.fullname`

File: incubator/busbar/templates/daemonset.yaml
Comment (unguiculus) [inline incubator/busbar/templates/daemonset.yaml]: Anything your chart installs should use `{{ template "busbar.fullname" . }}`. Use components to distinguish.

File: incubator/busbar/templates/daemonset.yaml
Comment (unguiculus) [inline incubator/busbar/templates/daemonset.yaml]: Make image configurable.

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: Structure everything by components.

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: Don't specify defaults for resources. We usually recommend to leave this a conscious choice of the user.

**Review by unguiculus — CHANGES_REQUESTED** [review]
I've just made a few comments. Try and apply them throughout your chart. Add components labels and make sure your get label selectors right adding `app`, `release`, and `component` labels.

Please read this: https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Here's a good example for a chart with multiple components: https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

File: incubator/busbar/templates/daemonset.yaml
Comment (waldman) [reply incubator/busbar/templates/daemonset.yaml]: Hmm... but this daemonset is not related to busbar. It's the kubernetes registry one.

Maybe it should be `kubeRegistry.name` instead `fullname`?

File: incubator/busbar/templates/svc.yaml
Comment (unguiculus) [inline incubator/busbar/templates/svc.yaml]: One file per resource, please.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Why don't you add a dependency to the registry chart?

File: incubator/busbar/templates/svc.yaml
Comment (waldman) [reply incubator/busbar/templates/svc.yaml]: Because we use a S3 based registry. The `docker-registry` is based on PVs. 

Re: one file per resource, I'm working on it! =))

Comment (unguiculus) [reply incubator/busbar/templates/svc.yaml]: S3 was recently added to the docker-registry chart. The chart is about to be moved to stable.

File: incubator/busbar/values.yaml
Comment (mattfarina) [inline incubator/busbar/values.yaml]: Instead of separating `repository` and `name` it should be called `repository` and have a value of `busbario/busbar`. The includes in other files should be updated as well.

Comment (waldman) [reply incubator/busbar/values.yaml]: Cool... Fixing it shortly! =)

**Review by mattfarina — APPROVED** [review]

**Comment by mattfarina** [comment]
I'm now leaving this for @unguiculus to come back around to. Want to make sure his issues are addressed in a way that he's good with.

**Comment by unguiculus** [comment]
Let me ask you again if you would consider adding a dependency to the Docker registry chart instead of installing it yourself. As mentioned it can now handle S3 as well. Also, the way you set up the registry is AWS-specific which doesn't seem ideal.

**Comment by waldman** [comment]
Hi @unguiculus,

I would rather not use the docker-registry dependency for now. Currently Busbar is AWS centric and we do have plans to make it GCE compatible in the near future so I would prefer to add the Docker Registry dependency once we roll out the GCE compatibility (what would also give us some more time to test/adjust the charts accordingly).

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: start with local case letter

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: name -> repository

File: incubator/busbar/values.yaml
Comment (unguiculus) [inline incubator/busbar/values.yaml]: name -> repository

File: incubator/busbar/README.md
Comment (unguiculus) [inline incubator/busbar/README.md]: Config table not correct.

File: incubator/busbar/templates/busbar-web-deployment.yaml
Comment (unguiculus) [inline incubator/busbar/templates/busbar-web-deployment.yaml]: Components should not use the name template. that's already in the `app` label.

`component: web`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by waldman** [comment]
Hi guys, any news on this? =)

**Comment by mattfarina** [comment]
@waldman It appears the chart is failing CI. The chart isn't working. Can you take a look at the CI issues and try to correct them? We cannot merge a PR when CI is failing and tend to be much slower on reviews if we know the chart is going to need to change to fix the CI issues.

**Pull request closed without merging**
